aiida/cmdline/groups/dynamic.py

```diff
@@ -68,27 +71,27 @@ def list_commands(self, ctx) -> list[str]:
         ])
         return sorted(commands)
 
-    def get_command(self, ctx, cmd_name):
+    def get_command(self, ctx: click.Context, cmd_name: str) -> click.Command | None:
         """Return the command with the given name.
 
         :param ctx: The :class:`click.Context`.
         :param cmd_name: The name of the command.
         :returns: The :class:`click.Command`.
         """
         try:
-            command = self.create_command(ctx, cmd_name)
+            command: click.Command | None = self.create_command(ctx, cmd_name)
```

Comment:
`create_command` returns `click.Command` so why is `command` typed as `click.Command | None`
```suggestion
            command: click.Command = self.create_command(ctx, cmd_name)
```

Reply:
Yes, but the `super().get_command(ctx, cmd_name)` below returns `click.Command | None` to the same variable. If I remove this annotation mypy complains

```python
aiida/cmdline/groups/dynamic.py:82: error: Incompatible types in assignment (expression has type "Optional[Command]", variable has type "Command")  [assignment]
```

I don't know why mypy complains, it should be able to infer this from the return type. Is there a better solution to this?

Reply:
Hmm weird, let's leave it then for now.